### DISCUSSION TITLE
Does `buildoss` produce a software licensed under BSL?

### DISCUSSION THREAD

**Discussion by tisonkun** [cockroachdb/cockroach#78558] [Q&A] [answered]
IIUC BSL is not an open-source software and `buildoss` implies that I can get an open-source licensed artifact for free use. 

However, it seems that `buildoss` will produce a software under BSL as all files doesn't licensed under CCL is default to BSL so that the usage is limited.

**Comment by rafiss** [comment, marked as answer]
Hi @tisonkun ! That sounds right; the makefile target would be more accurately named if it were called `buildbsl`.

> implies that I can get an open-source licensed artifact for free use.

Though just to clarify this, as long as you are not offering a commercial version of CockroachDB as a service, then you can still use the `buildoss` target for other uses. See [our blog post](https://www.cockroachlabs.com/blog/oss-relicensing-cockroachdb/) about it for more details.

**Reply by tisonkun** [reply]
Thanks for your answering @rafiss! Let me try to prepare a PR for changing the target.

Apart from that, I'd like to make sure some details about "offering a commercial version of CockroachDB as a service". Said I'm building a data warehouse service and using CockroachDB for saving metadata (if you heard about it, think of the role of FoundationDB to Firebolt), although I think directly depend on CockroachDB Serverless is a good idea, if I host my own CockroachDB cluster for such usage (internal access as a metadata service inside the data warehouse system, not expose crdb directly), is it violate the BSL terms?

**Reply by jordanlewis** [reply]
Hi @tisonkun, please refer to the guidelines here: https://www.cockroachlabs.com/docs/stable/licensing-faqs.html#what-constitutes-hosting-cockroachdb-as-a-service
